config/dynamic.c

```diff
@@ -514,17 +512,33 @@ run_dynamic_config(struct gk_config *gk_conf, struct gt_config *gt_conf,
 			"dyn_cfg: Failed to unlink(%s) - (%s)\n",
 			dy_conf->server_path, strerror(errno));
 		ret = -1;
-		goto free_path;
+		goto free_server_path;
 	}
 
+	dy_conf->lua_dy_base_dir = rte_malloc(
+		"lua_dy_base_dir", strlen(lua_dy_base_dir) + 1, 0);
```

Comment:
1. Rename `gt_strdup()` to `rte_strdup()`, move it to a proper location, and apply it here. Notice that `gt_strdup()` was introduced by a recently merged pull request.

2. I can't comment there, but apply `rte_strdup()` on `dy_conf->server_path` as well.